GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -568,12 +590,30 @@ private void PostFetchJob(List<string> packIndexes)
                         return;
                     }
 
-                    if (!this.gitObjects.TryWriteMultiPackIndex(this.context.Tracer, this.context.Enlistment, this.context.FileSystem))
+                    using (ITracer activity = this.context.Tracer.StartActivity("TryWriteMultiPackIndex", EventLevel.Informational, Keywords.Telemetry, metadata: null))
                     {
-                        this.context.Tracer.RelatedWarning(
-                            metadata: null,
-                            message: PostFetchTelemetryKey + ": Failed to generate multi-pack-index for new packfiles",
-                            keywords: Keywords.Telemetry);
+                        this.backgroundGitProcess = new GitProcess(this.context.Enlistment);
+
+                        if (this.stopping)
+                        {
+                            this.context.Tracer.RelatedWarning(
+                                metadata: null,
+                                message: PostFetchTelemetryKey + ": Not launching 'git multi-pack-index' because the mount is stopping",
+                                keywords: Keywords.Telemetry);
+                            return;
+                        }
+
+                        GitProcess.Result result = this.backgroundGitProcess.WriteMultiPackIndex(this.context.Enlistment.GitObjectsRoot);
+                        this.backgroundGitProcess = null;
+
+                        if (result.HasErrors)
+                        {
+                            this.context.Tracer.RelatedWarning(
+                                metadata: null,
+                                message: PostFetchTelemetryKey + ": Failed to generate multi-pack-index for new packfiles:" + result.Errors,

```

Comment:
Same here, record if we are stopping, so that we can be sure this was expected.

Reply:
Oops. I misread this the first time, and you are correct, we should still log this but track the stopping variable.